### PR TITLE
Make the tests run (and pass) in CI

### PR DESCRIPTION
**Pull request by andrewmarkle**
This makes the github actions run when you push up code. All the other code changes were rubocop related and I fixed all of those things. 

Also I removed ruby and rails versions that are end of life. 

Right now it fails on Rails versions 7.1 and 7.2 but we can add those in once we merge  https://github.com/Fullscript/yabeda-activejob/pull/16

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,25 +1,29 @@
+## 0.6.0 Unreleased
+- Fix rubocop rules and configure Github actions to run the specs for supported Rails / Ruby versions.
+- Remove support for end of life versions of Ruby and Rails.
+
 ## 0.5.0 - 2023-11-01
-- Add support for rails 7.1 Shout out to @bronislav for getting this done!! 
+- Add support for rails 7.1 Shout out to @bronislav for getting this done!!
 
-## 0.4.0 - 20220-11-04
+## 0.4.0 - 2022-11-04
```
Comment (ascullion-FS) [inline CHANGELOG.md:8]: Nice catch!

File: Gemfile [hunk 2]
```diff
@@ -5,3 +5,13 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 # Specify your gem's dependencies in yabeda-activejob.gemspec.
 gemspec
+
+group :development, :test do
```
Comment (ascullion-FS) [inline Gemfile:9]: Most of these are already in the `yabeda-activejob.gemspec` file. Would it make more sense to add the new dependencies there?

Comment (andrewmarkle) [reply Gemfile:9]: Rubocop yelled at me to not do that so I moved it out of there.  Also we need these here to run them in CI.

Comment (ascullion-FS) [reply Gemfile:9]: I read through some conversations and blog posts and I _think_ I understand the reasoning. I'll give this one last read and approve.

**Review by ascullion-FS — APPROVED** [review]

**Pull request merged**